### PR TITLE
[Discussion] New version of MessageQueue module

### PR DESCRIPTION
**Pull request by hgiemza**
I am pushing an "early access" version of changes in the MessageQueues module. It includes all the issues I mentioned in the email with possible solutions.

Not tested yet in a real environment as during the discussion we may come to a conclusion that something might need to be reverted to the previous version.

### CODE REVIEW

File: Resources/MessageQueue/test/Test_Utilities.py [hunk 1]
```diff
@@ -0,0 +1,168 @@
+"""
```
Comment (fstagni) [inline Resources/MessageQueue/test/Test_Utilities.py]: Please, just call this file "Test_MQ_Utilities.py" or Travis will get confused.

**Comment by fstagni** [comment]
Thanks! BTW these 2 new tests you have introduced are failing.

Anyway, this is is not the main point: please @atsareg and @zmathe review the changes here.
And this is interesting also for @wkrzemien

**Comment by hgiemza** [comment]
The test that are failing were prepared by Andrei for the previous version - I will fix everything when we agree how this module should look like.

**Comment by fstagni** [comment]
Updates in #3187

**Pull request closed without merging**